alf/algorithms/ddpg_algorithm.py

```diff
@@ -277,10 +277,13 @@ def actor_loss_fn(dqda, action):
             loss = 0.5 * losses.element_wise_squared_loss(
                 (dqda + action).detach(), action)
             loss = loss.sum(list(range(1, loss.ndim)))
-            if self.action_l2 > 0:
+            if self._action_l2 > 0:
                 assert action.requires_grad
-                loss += self.action_l2 * (torch.mean(
-                    torch.norm(action, dim=-1)**2))
+                assert action.ndim < 3, (
+                    "Action has ndim: {}.  Tensor.norm ".format(action.ndim) +
+                    "will incorrectly take the norm of the last dim.")
+                loss += self._action_l2 * (torch.mean(
```

Comment:
The shape of loss is [B]. We should add the action_l2 loss separately to each sample:
```
loss += self._action_l2 * (action * action).sum(list(range(1, action.ndim)))
```